### PR TITLE
[4.0] [com_newsfeeds] Parameterized queries (frontend)

### PR DESCRIPTION
**Pull request by SharkyKZ**
### Summary of Changes

Adds parameterized queries and cleans up SQL in frontend part of `com_newsfeeds`.

### Testing Instructions

Create some newsfeeds or install testing sample data.
Browse `com_newsfeeds` views in frontend.

### Expected result

Works like before.

### Documentation Changes Required

No.

### CODE REVIEW

File: components/com_newsfeeds/Model/CategoryModel.php [hunk 1]
```diff
@@ -143,61 +144,77 @@ public function getItems()
...
+			$nowDate = Factory::getDate()->toSql();
+
+			$query->extendWhere(
+				'AND',
+				[
+					$db->quoteName('a.publish_up') . ' IS NULL',
+					$db->quoteName('a.publish_up') . ' <= :nowDate1',
+				],
+				'OR'
+			)
```
Comment (Quy) [inline components/com_newsfeeds/Model/CategoryModel.php:194]: Indent?

File: components/com_newsfeeds/Model/NewsfeedModel.php [hunk 2]
```diff
@@ -88,43 +89,80 @@ public function &getItem($pk = null)
...
+					$archived  = (int) $archived;
+
+					$query->extendWhere(
+						'AND',
+						[
+							$db->quoteName('a.published') . ' = :published1',
+							$db->quoteName('a.published') . ' = :archived1',
+						],
+						'OR'
+					)
```
Comment (Quy) [inline components/com_newsfeeds/Model/NewsfeedModel.php:131]: Indent?

File: components/com_newsfeeds/Model/CategoryModel.php [hunk 3]
```diff
@@ -143,61 +144,77 @@ public function getItems()
...
 		{
-			$query->where('a.published = ' . (int) $state);
+			$state = (int) $state;
+			$query->where($db->quoteName('a.published') . ' = :state')
+				->bind(':state', $state, ParameterType::INTEGER);
 		}
 		else
 		{
-			$query->where('(a.published IN (0,1,2))');
+			$query->where($db->quoteName('a.published') . ' IN (0,1,2)');
```
Comment (Quy) [inline components/com_newsfeeds/Model/CategoryModel.php:179]: Use `whereIn`?

Comment (SharkyKZ) [reply components/com_newsfeeds/Model/CategoryModel.php:179]: No need for hardcoded values.

File: components/com_newsfeeds/Model/CategoryModel.php (hunk 1 above)
Comment (SharkyKZ) [reply components/com_newsfeeds/Model/CategoryModel.php:194]: Sorry, I don't see what to indent here. Looks correct to me.

Comment (Quy) [reply components/com_newsfeeds/Model/CategoryModel.php:194]: Indent lines 188-194 to align with the other `extendWhere`.

Comment (SharkyKZ) [reply components/com_newsfeeds/Model/CategoryModel.php:194]: It is correct as it is, according to our coding standards. But if you want we can move `extendWhere()` call to next line like this:
```
$query
	->extendWhere(
```
So it aligns with the rest.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 75deb4437109981158cdbb5f881f697f36ab37b1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27207">issues.joomla.org/tracker/joomla-cms/27207</a>.</sub>

**Comment by jwaisner** [comment]
I have tested this item :white_check_mark: successfully on 75deb4437109981158cdbb5f881f697f36ab37b1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27207">issues.joomla.org/tracker/joomla-cms/27207</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27207">issues.joomla.org/tracker/joomla-cms/27207</a>.</sub>

**Comment by HLeithner** [comment]
@jwaisner I'm a bit impressed by your testing speed can you please elaborate how you test? The reason I ask is because you tested 3 PRs in 5 minutes including a complete component. So it seams I could learn something from you.

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**